### PR TITLE
Feature: Update build pipeline

### PR DESCRIPTION
**Pull request by tenequm**

### CODE REVIEW

File: devops/dockerfiles/node-new/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+FROM base AS builder
+COPY --from=planner /app/recipe.json recipe.json
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --check --release --no-default-features -p mangata-node
+
+# Build application
+COPY . .
+RUN cargo build --release -p mangata-node --no-default-features --features=mangata-rococo,mangata-kusama \
+  && mv /app/target/release/mangata-node /app/node \
```
Comment (goncer) [inline devops/dockerfiles/node-new/Dockerfile:18]: which one is the output of /release/mangata-node?  kusama or rococo?

File: devops/dockerfiles/node-new/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --check --release --no-default-features -p mangata-node
+
+# Build application
+COPY . .
+RUN cargo build --release -p mangata-node --no-default-features --features=mangata-rococo,mangata-kusama,runtime-benchmarks \
+  && mv /app/target/release/mangata-node /app/mangata-node \
+  && mv /app/target/release/wbuild/mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm /app/ \
+  && mv /app/target/release/wbuild/mangata-kusama-runtime/mangata_kusama_runtime.compact.compressed.wasm /app/ \
+  && rm -rf /app/target /usr/local/cargo /root/.cache
```
Comment (mateuszaaa) [inline devops/dockerfiles/node-new/Dockerfile:21]: :+1:

File: devops/dockerfiles/node-new/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# Build application
+COPY . .
+RUN cargo build --release -p mangata-node --no-default-features --features=mangata-rococo,mangata-kusama,runtime-benchmarks \
+  && mv /app/target/release/mangata-node /app/mangata-node \
+  && mv /app/target/release/wbuild/mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm /app/ \
+  && mv /app/target/release/wbuild/mangata-kusama-runtime/mangata_kusama_runtime.compact.compressed.wasm /app/ \
+  && rm -rf /app/target /usr/local/cargo /root/.cache 
+
+FROM debian:buster
```
Comment (mateuszaaa) [inline devops/dockerfiles/node-new/Dockerfile:23]: are we 100% sure that `debian:buster` is base image also for `paritytech/ci-linux:production`

File: devops/dockerfiles/node-new/Dockerfile (hunk 1 above)
Comment (mateuszaaa) [reply devops/dockerfiles/node-new/Dockerfile]: that seems to be true at the moment, right @iStrike7 , do you see any problem with that ?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -141,204 +141,65 @@ jobs:
...
-        name: Upload wasm artifact
-        uses: actions/upload-artifact@v2
-        if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip-build') && github.event.inputs.skipBuild != 'true' }}
-        with:
-          name: kusama-wasm-${{ needs.init.outputs.GLOBAL_VERSION }}
-          path: |
-            ./docker-cargo/release/wbuild/mangata-kusama-runtime/mangata_kusama_runtime.compact.compressed.wasm
-            ./docker-cargo/release/wbuild/mangata-kusama-runtime/mangata_kusama_runtime.compact.compressed.wasm.md5
+          docker pull ${{ env.NODE_IMAGE }}
+          docker run --rm ${{ env.NODE_IMAGE }} benchmark pallet --chain kusama-mainnet --execution wasm --wasm-execution compiled --extrinsic '*' --pallet 'pallet-xyk' --template ./templates/module-weight-template.hbs
```
Comment (mateuszaaa) [inline .github/workflows/ci.yml:195]: why sometime we use docker image and sometimes `docker-cargo.sh` are we maintain both configurations ?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -141,204 +141,65 @@ jobs:
...
-        if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip-build') && github.event.inputs.skipBuild != 'true' }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
         with:
-          name: rococo-wasm-${{ needs.init.outputs.GLOBAL_VERSION }}
-          path: |
-            ./docker-cargo/release/wbuild/mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm
-            ./docker-cargo/release/wbuild/mangata-rococo-runtime/mangata_rococo_runtime.compact.compressed.wasm.md5
+          push: true
```
Comment (mateuszaaa) [inline .github/workflows/ci.yml:175]: so now we are going to push docker image with every pr ?

**Review by mateuszaaa — CHANGES_REQUESTED** [review]
please stash changes into single commit or rework commits history

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -354,23 +215,11 @@ jobs:
...
-        with:
-          name: rococo-docker-${{ needs.init.outputs.GLOBAL_VERSION }}
-          path: /tmp
-      -
-        name: Load Docker image
-        if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip-build') && github.event.inputs.skipBuild != 'true' }}
-        run: |
-          docker load --input /tmp/rococo-docker-${{ needs.init.outputs.GLOBAL_VERSION }}.tar
-          docker image ls -a
+        run: docker pull ${{ env.PARACHAIN_DOCKER_IMAGE }}
```
Comment (goncer) [inline .github/workflows/ci.yml:222]: why this? are not we using `"mangatasolutions/mangata-node:rococo-${{ needs.init.outputs.GLOBAL_VERSION }}"` ?

File: devops/dockerfiles/node-new/Dockerfile (hunk 3 above)
Comment (mateuszaaa) [reply devops/dockerfiles/node-new/Dockerfile:24]: i think we should push it to our dockerhub, in case if they remove it for some reason :thinking:

**Review by mateuszaaa — APPROVED** [review]
lgtm

**Review by goncer — APPROVED** [review]

**Pull request merged**